modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/WikiPageResourceImpl.java

```diff
@@ -366,7 +366,7 @@ protected String getPermissionCheckerResourceName(Object id) {
 		throws Exception {
 
 		com.liferay.wiki.model.WikiPage wikiPage =
-			_wikiPageLocalService.fetchWikiPageByExternalReferenceCode(
+			_wikiPageLocalService.fetchLatestPageByExternalReferenceCode(
```

Comment:
I think there are two problems here.

First, we should call getLatestPageByExternalReferenceCode instead of fetch, to delegate throwing the NoSuchElement exception into the Service.

Second, this method is called from delete, from where it would be correct to call the LocalService as there is a permission check to delete the WikiPage after the call, and from get, where there is no permission check, so I think in the second case we should call the Service instead of the LocalService to check permissions before getting a WikiPage

Reply:
^ 👏

Reply:
Done. I delegated throwing the exception to the service and called the RemoteService to check permissions.